src/main/java/com/cognite/sa/beam/replicate/ReplicateLabels.java

```diff
@@ -0,0 +1,162 @@
+        @Validation.Required
+        ValueProvider<String> getCdfTargetSecret();
+
+        void setCdfTargetSecret(ValueProvider<String> value);
+
+        @Description("The CDF source host name. The default value is https://api.cognitedata.com")
+        @Default.String("https://api.cognitedata.com")
+        ValueProvider<String> getCdfTargetHost();
+
+        void setCdfTargetHost(ValueProvider<String> value);
+
+    }
+
+    /**
+     * Setup the main pipeline structure and run it:
+     * - Read the config settings
+     * - Read the datasets from both source and target (in order to map the labels to the proper dataset)
+     * - Replicate the labels
+     *
+     * @param options
+     * @return
+     * @throws IOException
+     */
+    private static PipelineResult runReplicateLabels(ReplicateLabelsOptions options) throws IOException {
+        /*
+        Build the project configuration (CDF tenant and api key) based on:
+        - api key from Secret Manager
+        - CDF api host
+         */
+        GcpSecretConfig sourceSecret = GcpSecretConfig.of(
+                ValueProvider.NestedValueProvider.of(options.getCdfInputSecret(), secret -> secret.split("\\.")[0]),
+                ValueProvider.NestedValueProvider.of(options.getCdfInputSecret(), secret -> secret.split("\\.")[1]));
+        ProjectConfig sourceConfig = ProjectConfig.create()
+                .withApiKeyFromGcpSecret(sourceSecret)
+                .withHost(options.getCdfInputHost());
+        GcpSecretConfig targetSecret = GcpSecretConfig.of(
+                ValueProvider.NestedValueProvider.of(options.getCdfTargetSecret(), secret -> secret.split("\\.")[0]),
+                ValueProvider.NestedValueProvider.of(options.getCdfTargetSecret(), secret -> secret.split("\\.")[1]));
+        ProjectConfig targetConfig = ProjectConfig.create()
+                .withApiKeyFromGcpSecret(targetSecret)
+                .withHost(options.getCdfTargetHost());
+
+        Pipeline p = Pipeline.create(options);
+
+        /*
+        Read Labels from source and parse them:
+        - Remove system fields ( last updated time)
+        - Translate data set id link
+
+        Write the prepared labels to target
+         */
+        PCollection<Label> labelPCollection = p
+                .apply("Build basic query", Create.of(RequestParameters.create()))
+                .apply("Read source Labels", CogniteIO.readAllLabels()
+                        .withProjectConfig(sourceConfig)
+                        .withHints(Hints.create())
+                        .withReaderConfig(ReaderConfig.create()
+                                .withAppIdentifier(appIdentifier)))
+                .apply("Prepare Labels", ParDo.of(new PrepareLabels()))
+                .apply("Write target Labels", CogniteIO.writeLabels()
+                        .withProjectConfig(targetConfig)
+                        .withHints(Hints.create()
+                                .withWriteShards(20))
```

Comment:
Let's downscale the number of shards. The labels data set is usually quite small, so we can go with 2 shards or so.

Reply:
ok